tools/release

```diff
@@ -172,35 +185,101 @@ done
 assert_clean || die 'tools/build should not dirty the working directory'
 
 # 4. Push the branch so release-build-scripts repository
-$nop git push origin $branch
+if $push; then
+    $nop git push origin $branch
+fi
 
 # 5. Trigger toolkit packaging actions in release-build-scripts repository.
 branch_commit=`git log -1 --pretty=format:%h`
-$nop gh workflow run toolkit-package.yml \
+if $push; then
+    $nop gh workflow run toolkit-package.yml \
         -R timescale/release-build-scripts \
         -r main \
         -f version=$VERSION \
         -f commit-id=$branch_commit \
         -f upload-artifacts=true
+fi
 
 # 6. Tag the release.
 $nop git tag $VERSION
-$nop git push origin $VERSION
+if $push; then
+    $nop git push origin $VERSION
+fi
 
 # 7. Prepare the main branch for the next release cycle.
 # Github action gives us a shallow checkout which we must deepen before we can push changes.
 $nop git fetch --deepen=2147483647 origin main
 $nop git checkout main
 
 # 7a. Set toolkit version to released version with '-dev' appended.
-cargo set-version $dry_run_flag -p timescaledb_toolkit $VERSION-dev
-assert_dirty || die "failed to set toolkit version to $VERSION-dev in $TOML"
+major=${VERSION%%.*}
+minpat=${VERSION#*.}
+minor=${minpat%.*}
+DEV_VERSION=$major.$(( minor + 1 )).0-dev
+cargo set-version $dry_run_flag -p timescaledb_toolkit $DEV_VERSION
+assert_dirty || die "failed to set toolkit version to $DEV_VERSION in $TOML"
 start_commit $TOML
 # Update cargo.lock - this form of cargo update doesn't update dependency versions.
 $nop cargo update -p timescaledb_toolkit
-assert_dirty || die "failed to set toolkit version to $VERSION-dev in Cargo.lock"
+assert_dirty || die "failed to set toolkit version to $DEV_VERSION in Cargo.lock"
 commit Cargo.lock
-finish_commit "start $VERSION-dev"
+
+# 7b. Update Changelog.md .
+branch_commit_date=`git log -1 --pretty=format:%as $branch_commit`
+$nop ed Changelog.md <<EOF
+/^## Next Release (Date TBD)/
+a
+
+#### New experimental features
+
+#### Bug fixes
+
+#### Other notable changes
+
+#### Shout-outs
+
+**Full Changelog**: [TODO]
+
+## [$VERSION](https://github.com/timescale/timescaledb-toolkit/releases/tag/$VERSION) ($branch_commit_date)
+.
+wq
+EOF
+assert_dirty || die 'failed to update Changelog.md for next release'
+commit Changelog.md
+finish_commit "start $DEV_VERSION"
 $nop git show
 
-$nop git push
+# TODO Create a pull request instead?
+if $push; then
+    $nop git push origin main
+fi
+
+# File issue for release tasks that are not yet automated.
+gh issue create -R timescale/timescaledb-toolkit -F- -t "Release $VERSION" <<EOF
```

Comment:
I think we would only want to do this if `$push` is set? And also it should have a `$nop` at the start?

Reply:
Right on both counts!  I'll address that later today.  Thanks!